packages/sdk/src/filter/filter.ts

```diff
@@ -1,297 +1,259 @@
 import { ConnectionManager, FilterCore } from "@waku/core";
 import type {
   Callback,
-  CreateSubscriptionResult,
   FilterProtocolOptions,
-  IAsyncIterator,
   IDecodedMessage,
   IDecoder,
   IFilter,
-  ILightPush,
-  Libp2p,
-  PubsubTopic,
-  SubscribeResult,
-  Unsubscribe
+  Libp2p
 } from "@waku/interfaces";
-import { NetworkConfig, ProtocolError } from "@waku/interfaces";
-import {
-  ensurePubsubTopicIsConfigured,
-  groupByContentTopic,
-  Logger,
-  shardInfoToPubsubTopics,
-  toAsyncIterator
-} from "@waku/utils";
+import { WakuMessage } from "@waku/proto";
+import { Logger } from "@waku/utils";
 
 import { PeerManager } from "../peer_manager/index.js";
 
 import { Subscription } from "./subscription.js";
-import { buildConfig } from "./utils.js";
+import { FilterConstructorParams } from "./types.js";
 
-const log = new Logger("sdk:filter");
+const log = new Logger("sdk:next-filter");
 
-type FilterConstructorParams = {
-  connectionManager: ConnectionManager;
-  libp2p: Libp2p;
-  peerManager: PeerManager;
-  lightPush?: ILightPush;
-  options?: Partial<FilterProtocolOptions>;
-};
+type PubsubTopic = string;
 
 export class Filter implements IFilter {
-  public readonly protocol: FilterCore;
+  private readonly libp2p: Libp2p;
+  private readonly protocol: FilterCore;
```

Comment:
nit: it's not immediately clear that this the singleton instance of `FilterCore` used by all subscriptions

Reply:
I am not sure I understand the comment
`protocol` in context of `Filter SDK` can mean only one thing imho